Source/Models/APIClient.swift

```diff
@@ -4,34 +4,34 @@ import Result
 
 public struct APIClient {
   private let requestPerformer: RequestPerformer
+  public let deserializer: Deserializer
 
-  public init(requestPerformer: RequestPerformer = NetworkRequestPerformer()) {
+  public init(requestPerformer: RequestPerformer = NetworkRequestPerformer(), deserializer: Deserializer = JSONDeserializer()) {
     self.requestPerformer = requestPerformer
+    self.deserializer = deserializer
   }
 }
 
 extension APIClient: Client {
   public func performRequest<T: Request>(request: T, completionHandler: Result<T.ResponseObject, T.ResponseError> -> Void) -> NSURLSessionDataTask {
     return requestPerformer.performRequest(request.build()) { result in
-      let object = (result >>- deserialize >>- request.parse).mapError(request.transformError)
+
+      let object = (result
+        >>- self.clearResponse
+        >>- self.deserializer.deserialize(request)
+        >>- request.parse).mapError(request.transformError)
```

Comment:
what do you think about dropping the closing paren and `mapError` onto the next line?